hnn_core/network.py

```diff
@@ -1253,23 +1253,70 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
+    def _clear_connectivity(self, src_types):
+        """Remove connections with src_type in Network.connectivity.
+
+        Parameters
+        ----------
+        src_types : list
+            Source types of connections to be cleared
+
         """
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] in self.external_drives.keys():
-                connectivity.append(conn)
-        self.connectivity = connectivity
-
-    def clear_drives(self):
-        """Remove all drives defined in Network.connectivity"""
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] not in self.external_drives.keys():
-                connectivity.append(conn)
-        self.external_drives = dict()
-        self.connectivity = connectivity
+        _validate_type(src_types, list, 'src_types', 'list, drives, local')
+        # Finding connection indices to be deleted
+        conn_idxs = list()
+        for src_type in src_types:
+            conn_idxs.extend(pick_connection(self, src_gids=src_type))
+
+        # Deleting the indices
+        for conn_idx in sorted(conn_idxs, reverse=True):
+            del self.connectivity[conn_idx]
+
+    def clear_connectivity(self, src_types='all'):
+        """Clear connections between cells of the local network
```

Comment:
@rythorpe is this what you intended? A "dangling" cell population will be problematic even with local network connectivity ... I'm wondering if we should make `clear_connectivity` private altogether until it has clarified in our minds how to do it right

Reply:
Yep! I don't see how this could be problematic. If you think we should prevent users from selectively removing local network connections, perhaps we can just get rid of the `src_types` arg and make this function clear all local network connections (which I guess is more consistent with the name of this method anyway).

Reply:
but `clear_drives` needs the `src_types` argument ...

Reply:
> but `clear_drives` needs the `src_types` argument ...

why?